sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2224,6 +2348,18 @@ object EliminateSubqueryAliases extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes [[View]] operators from the plan. The operator is respected till the end of analysis
+ * stage because we want to see which part of a analyzed logical plan is generated from a view.
+ */
+object EliminateView extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    // If the child of a view is empty, we will throw an AnalysisException later in
+    // `checkAnalysis`.
+    case View(_, output, Some(child)) => Project(output, child)
```

Comment:
I am not sure the project is needed at all (the child should output the attributes anyway). Also make sure we check if the child outputs the required attributes (for example in checkanalysis).